fix: improve San'layn implementation for death knights

johnnylam88 opened the pull request:
Improve the San'layn hero talents implementation from 11.0.5 for death knights, and catch up to some talent changes for death knights from 11.0.

johnnylam88 committed:
fix: Essence of the Blood Queen stacks up to 5 (+2)

johnnylam88 committed:
fix: apply Infliction of Sorrow after Gift of the San'layn expires

johnnylam88 committed:
fix: improve San'layn hero talents for blood death knight

johnnylam88 committed:
fix: Empower Rune Weapon is a frost-only talent

Remove Empower Rune Weapon references for blood and unholy death knights
as it is a Frost-only talent since 11.0 (TWW).

Hekili commented on TheWarWithin/DeathKnightBlood.lua:
```diff
@@ -1187,7 +1185,14 @@ spec:RegisterHook( "reset_precast", function ()
+
+    if IsActiveSpell( 433899 ) or IsActiveSpell( 433895 ) then
+        applyBuff( "vampiric_strike" )
+        class.abilities[ 433895 ] = class.abilities.heart_strike
```
Why here?  This normally goes in the `copy` line of the ability model (`RegisterAbilities`).

johnnylam88 replied:
You're right, that should be enough. I was mostly just copy-pasting from the unholy implementation for speed.

johnnylam88 replied:
I'll just delete that assignment and verify everything still works.

Hekili reviewed (changes requested):
One small check.

johnnylam88 committed:
fix: remove unnecessary assignment

`vampiric_strike` is already copied from `heart_strike`, so setting
`class.abilities[433895]` to match `heart_strike` is unnecessary.

johnnylam88 commented:
I've pushed one small commit to remove that unnecessary assignment. Testing on the target dummy shows that it's working correctly.
